app/models/chat.rb

```diff
@@ -6,4 +6,16 @@ class Chat < ApplicationRecord
   def users
     [user_a, user_b]
   end
+
+  def unread_messages_count(user)
+    messages.where(seen: false).where.not(user: user).count
```

Comment:
Why do you user a `where.not(user: user)`
I would suggest that if you pass a parameter to get the count, that it is the user you want to get the count (and not the other).